src/Npgsql/PostgresDatabaseInfo.cs

```diff
@@ -201,161 +203,139 @@ internal async Task<List<PostgresType>> LoadBackendTypes(NpgsqlConnector conn, N
             if (timeout.IsSet)
                 commandTimeout = (int)timeout.CheckAndGetTimeLeft().TotalSeconds;
 
-            using var batch = new NpgsqlBatch(conn)
-            {
-                BatchCommands =
-                {
-                    new("SELECT version()"),
-                    new(GenerateLoadTypesQuery(SupportsRangeTypes, conn.Settings.LoadTableComposites)),
-                    new(GenerateLoadCompositeTypesQuery(conn.Settings.LoadTableComposites)),
-                }
-            };
+            var batchQuery = new StringBuilder("SELECT version();")
+                .Append(GenerateLoadTypesQuery(SupportsRangeTypes, conn.Settings.LoadTableComposites))
+                .Append(GenerateLoadCompositeTypesQuery(conn.Settings.LoadTableComposites));
 
             if (SupportsEnumTypes)
-                batch.BatchCommands.Add(new(GenerateLoadEnumFieldsQuery(HasEnumSortOrder)));
-
-            batch.Timeout = commandTimeout;
-            batch.AllResultTypesAreUnknown = true;
+                batchQuery.Append(GenerateLoadEnumFieldsQuery(HasEnumSortOrder));
 
+            conn.WriteBuffer.Timeout = TimeSpan.FromSeconds(commandTimeout);
             timeout.CheckAndApply(conn);
-            var reader = async ? await batch.ExecuteReaderAsync() : batch.ExecuteReader();
-            try
-            {
-                var byOID = new Dictionary<uint, PostgresType>();
 
-                // First the PostgreSQL version
-                if (async)
-                {
-                    await reader.ReadAsync();
-                    LongVersion = reader.GetString(0);
-                    await reader.NextResultAsync();
-                }
-                else
+            var q = batchQuery.ToString();
+            Debug.WriteLine($"Type loading query:{Environment.NewLine}{q}");
+            await conn.WriteQuery(q, async);
+            await conn.Flush(async);
+            var byOID = new Dictionary<uint, PostgresType>();
+            var buf = conn.ReadBuffer;
+
+            // First read the PostgreSQL version
+            var rowDescription = Expect<RowDescriptionMessage>(await conn.ReadMessage(async), conn);
+            Debug.Assert(rowDescription.Count == 1, "The 'version query' should return exactly one column.");
+
+            // We read the message in non-sequential mode which buffers the whole message.
+            // There is no need to ensure data within the message boundaries
+            Expect<DataRowMessage>(await conn.ReadMessage(async), conn);
+            buf.Skip(2);
+            LongVersion = buf.ReadString(buf.ReadInt32());
+            Expect<CommandCompleteMessage>(await conn.ReadMessage(async), conn);
+
+            // Then load the types
+            rowDescription = Expect<RowDescriptionMessage>(await conn.ReadMessage(async), conn);
+            Debug.Assert(rowDescription.Count == 6, "The 'type loading query' should return exactly 6 columns.");
```

Comment:
Do we need this? It seems to check that PostgreSQL is doing its job correctly, which we generally don't need to do (plus we'll need to update it when the query above changes). Same for similar assertions below.

Reply:
No, it's definitely unnecessary.
I've put it in to help catching problems after changing the query but that's probably really not worthwhile.